File: demo/hello-scripts/src/main/webapp/WEB-INF/erb/render.rb
require 'erb'
require 'ostruct'
require 'java'
# https://github.com/sdeleuze/spring-framework/blob/SPR-12266/spring-webmvc/src/test/resources/org/springframework/web/servlet/view/script/erb/render.rb
# Renders an ERB template against a hashmap of variables.
# template should be a Java InputStream
def render(template, variables)
  context = OpenStruct.new(variables).instance_eval do
    variables.each do |k, v|
      instance_variable_set(k, v) if k[0] == '@'
    end

    def partial(partial_name, options={})
      new_variables = marshal_dump.merge(options[:locals] || {})
      Java::Pavo::ERB.render(partial_name, new_variables)
    end

    binding
  end
  ERB.new(template).result(context);
end

File: demo/hello-scripts/src/main/webapp/WEB-INF/react/render.js
// https://github.com/sdeleuze/spring-framework/blob/SPR-12266/spring-webmvc/src/test/resources/org/springframework/web/servlet/view/script/react/render.js

function render(template, model) {
    // Create a real Javascript Object from the model Map
    var data = {};
    for(var k in model) data[k]=model[k];
    var element = React.createElement(eval(template), data);
    // Should use React.renderString in production
    return React.renderToStaticMarkup(element);
}

function renderJsx(template, model) {
    var jsTemplate = JSXTransformer.transform(template).code;
    return render(jsTemplate, model);
}